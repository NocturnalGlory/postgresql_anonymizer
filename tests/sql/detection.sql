
BEGIN;

CREATE EXTENSION IF NOT EXISTS anon CASCADE;

-- INIT

SELECT anon.load();


CREATE TABLE customer (
  id SERIAL,
  firstname TEXT,
  last_name TEXT,
  "CreditCard" TEXT
);

CREATE TABLE vendor (
  employee_id INTEGER UNIQUE,
  "Firstname" TEXT,
  lastname TEXT,
  phone_number TEXT,
  birth DATE
);


SELECT anon.detect() IS NOT NULL LIMIT 1;

SELECT count(*) = 5  FROM anon.detect('fr_FR');


ROLLBACK;
