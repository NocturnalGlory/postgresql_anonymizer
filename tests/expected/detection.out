BEGIN;
CREATE EXTENSION IF NOT EXISTS anon CASCADE;
NOTICE:  installing required extension "tsm_system_rows"
NOTICE:  installing required extension "ddlx"
-- INIT
SELECT anon.load();
 load 
------
 t
(1 row)

CREATE TABLE customer (
  id SERIAL,
  firstname TEXT,
  last_name TEXT,
  "CreditCard" TEXT
);
CREATE TABLE vendor (
  employee_id INTEGER UNIQUE,
  "Firstname" TEXT,
  lastname TEXT,
  phone_number TEXT,
  birth DATE
);
SELECT anon.detect() IS NOT NULL LIMIT 1;
 ?column? 
----------
 t
(1 row)

SELECT count(*) = 5  FROM anon.detect('fr_FR');
 ?column? 
----------
 t
(1 row)

ROLLBACK;
